report_gen/README.md

```diff
@@ -0,0 +1,157 @@
+# Report generation
+
+## Using graphgen.py
+
+### Arguments:  
+	-f, --csv_file - CSV file(s) to process ( Required )  
+	-o, --out_dir - Output directory for the graphs ( Default: ./ )  
+	-x, --x_axes_str - Create graphs with x axes being the columns containing these strings ( Default: First column of first file )  
+	--plot_col - Only plot the columns with these names while plotting across csv files ( Default: All columns )  
+	-e, --extension - Specify the extension of the graphs ( Default: svg )  
+	-c, --categories - Plot by category ( Default: no category )   
+
+### Basic functionality
+
+Read supplied csv files, create graphs from the columns they have in common, and store them. As default, the x axis is the first column in the first file.  
+
+Ignores the first line of the csv file, as it commonly holds the name of the command which generated it.
+
+Output filenames are of the format: all:x_axis-y_axis.extension
+
+####CSV files used in example:
+
+    a.csv:                 b.csv:                 c.csv:
+    
+    firstline              firstline              firstline
+    x0, y1, y2, y3         y1, x0, y2, z4         x0, y1, y3, z4
+    1,  2,  8,  4          1,  1,  9,  5          1,  4,  8,  6
+    2,  4,  4,  3          2,  3,  5,  4          2,  6,  4,  5
+    3,  8,  2,  2          4,  7,  3,  3          3,  7,  2,  4
+
+As seen from the structure of the csv files:  
+1. The order of columns can be different from file to file  
+2. Not all files have to contain the same columns
+
+#### Example usage:
+    python graphgen.py -f example/a.csv example/b.csv example/c.csv -o example/figs/graphgen
+
+#### Generated graphs:  
+
+|example/figs/graphgen/all:x0-y1.svg|example/figs/graphgen/all:x0-y2.svg
+:-------------------------:|:-------------------------:
+<img src="./example/figs/graphgen/all:x0-y1.svg" width="300" height="auto"> | <img src="./example/figs/graphgen/all:x0-y2.svg" width="300" height="auto">
+
+|example/figs/graphgen/all:x0-y1.svg|example/figs/graphgen/all:x0-y2.svg
+:-------------------------:|:-------------------------:
+<img src="./example/figs/graphgen/all:x0-y3.svg" width="300" height="auto"> | <img src="./example/figs/graphgen/all:x0-z4.svg" width="300" height="auto">
+
+### Features
+#### 1. Plot by category
+
+Plots graphs by category: for each test separately, parses columns for search strings and plots them with a common name on the y axis. The search string and the name are input arguments in dict form: '{"searchstring":"name"}'. 
```

Comment:
For me, it was quite difficult to understand what is the category and what is "only plot specific columns". If I understand correctly. (1) is more like filtering columns with some pattern and (3) sounds to me like grouping the same columns from multiple files to one graph.

Reply:
(1) is filtering columns with a specific pattern in each file and plotting them into a single graph. So for 3 files, 3 graphs are generated for each category.

As default, columns with the same name are plotted to a graph, for all columns found in the files. So for N files and 4 unique column names we get 4 graphs.
(3) limits these columns to those which contain some pattern. So from the 4 column names, only the ones containing "string" have a graph generated.

I will try to improve the descriptions of (1) and (3), but I am unsure how to best explain this.